From ef4c0b2682415cd257c04c813d2eab8028eb1746 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Fri, 1 Mar 2024 02:03:33 +0000
Subject: [PATCH 16/16] devlink: remove devlink features

This commit accommodates commit fb8421a94c56 ("devlink: remove
devlink features"), which removes devlink_set_features() function.
This commit was merged in the v6.4 development cycle and causes
compilation failures with the v6.6 kernel, which are listed below.

We accommodate this commit by undefine HAVE_DEVLINK_SET_FEATURES
for the v6.6 kernel. It is also align with the function of
ice_devlink_register() in the ice driver of the upstream code.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=fb8421a94c56

Devlink features were introduced to disallow devlink reload calls of
userspace before the devlink was fully initialized. The reason for this
workaround was the fact that devlink reload was originally called
without devlink instance lock held.

However, with recent changes that converted devlink reload to be
performed under devlink instance lock, this is redundant so remove
devlink features entirely.

src/ice_devlink.c:1034:2: error: implicit declaration of function \
devlink_set_features; did you mean netif_skb_features? \
[-Werror=implicit-function-declaration]
 1034 |  devlink_set_features(devlink, DEVLINK_F_RELOAD);

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/kcompat_std_defs.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kcompat_std_defs.h b/src/kcompat_std_defs.h
index a48fe09..9b523ec 100644
--- a/src/kcompat_std_defs.h
+++ b/src/kcompat_std_defs.h
@@ -268,7 +268,9 @@
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(5,16,0))
 #else /* >= 5.16.0 */
 #undef HAVE_PASID_SUPPORT
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6,6,0))
 #define HAVE_DEVLINK_SET_FEATURES
+#endif
 #define HAVE_DEVLINK_NOTIFY_REGISTER
 #undef HAVE_DEVLINK_RELOAD_ENABLE_DISABLE
 #undef HAVE_DEVLINK_PARAMS_PUBLISH
-- 
2.43.0

